certification/internal/shell/has_minimal_vulns.go

```diff
@@ -1,15 +1,96 @@
 package shell
 
 import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
-	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/errors"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/utils/file"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	ovalFilename = "rhel-8.oval.xml.bz2"
+	ovalUrl      = "https://www.redhat.com/security/data/oval/v2/RHEL8/"
+	reportFile   = "vuln.html"
 )
 
 type HasMinimalVulnerabilitiesCheck struct{}
 
 func (p *HasMinimalVulnerabilitiesCheck) Validate(image string) (bool, error) {
-	return false, errors.ErrFeatureNotImplemented
+
+	ovalFileUrl := fmt.Sprintf("%s%s", ovalUrl, ovalFilename)
+	dir, err := ioutil.TempDir(".", "oval-")
+
+	if err != nil {
+		log.Error("Unable to create temp dir", err)
+		return false, err
+	}
+	log.Debugf("Oval file dir: %s", dir)
+	defer os.RemoveAll(dir)
+
+	ovalFilePath := filepath.Join(dir, ovalFilename)
+	log.Debugf("Oval file path: %s", ovalFilePath)
+
+	err = file.DownloadFile(ovalFilePath, ovalFileUrl)
+	if err != nil {
+		log.Error("Unable to download Oval file", err)
+		return false, err
+	}
+	// get the file name
+	r := regexp.MustCompile(`(?P<filename>.*).bz2`)
+	ovalFilePathDecompressed := filepath.Join(dir, r.FindStringSubmatch(ovalFilename)[1])
+
+	err = file.Unzip(ovalFilePath, ovalFilePathDecompressed)
+	if err != nil {
+		log.Error("Unable to unzip Oval file: ", err)
+		return false, err
+	}
+
+	numOfVulns, err := p.numberOfVulnerabilities(image, ovalFilePathDecompressed)
+	if err != nil {
+		return false, err
+	}
+
+	log.Debugf("The number of found vulnerabilities: %d", numOfVulns)
+	if numOfVulns > 0 {
+		return false, nil
+	}
+
+	return true, nil
 }
+
+func (p *HasMinimalVulnerabilitiesCheck) numberOfVulnerabilities(image string, ovalFilePathDecompressed string) (int, error) {
```

Comment:
Can you add a comment here detailing the high level logic? Not strictly required, but a bulk of the work is here, so I think it'll help us when we come back to it in the future.

Reply:
Please let me know if it is enough details

Reply:
@samira-barouti just a nit, but one more thing here - it's idiomatic in Go to start function comments with the function name, example:

```go
// numberofVulnerabilities runs the ...
func (p *HasMinimalVulnerabilitiesCheck) numberOfVulnerabilities(image string, ovalFilePathDecompressed string) (int, error) {
...
}
```

Could you swap out `This method` for the method name?

For unexported methods, I don't believe it matters much, but for exported methods - this gets translated into the godoc for that library (which then gets published to pkg.go.dev, etc).

reference: https://blog.golang.org/godoc

Reply:
Of course! Will update the method's comment.